### PR TITLE
Add --on-batch-error cli hook for custom batch errors handling

### PR DESCRIPTION
**Pull request by nwlunatic**

### CODE REVIEW

File: lib/parallel_cucumber/worker.rb [hunk 1]
```diff
@@ -153,12 +154,43 @@ def run_batch(env, queue_tracker, results, running_total, tests)
...
+        }.to_json
+        File.write(error_file, batch_error_info)
+        command = "#{@on_batch_error} #{error_file}"
+        return Helper::Command.exec_command(
+            batch_env, 'on_batch_error', command, @logger, @log_decoration, timeout: @batch_timeout, capture: true
+        )
+      rescue
+        message = 'on-batch-error failed: quitting immediately'
+        @logger.error(message)
+        raise message
```
Comment (bayandin) [inline lib/parallel_cucumber/worker.rb]: Print error message and do not fail worker.

File: lib/parallel_cucumber/worker.rb [hunk 2]
```diff
@@ -153,12 +154,43 @@ def run_batch(env, queue_tracker, results, running_total, tests)
...
+    end
+
+    def on_batch_error(batch_env, batch_id, error_file, tests, error)
+      unless @on_batch_error
+        return 'default no-op on_batch_error'
+      end
+
+      begin
+        error_info = {
+            'class'     => error.class,
```
Comment (bayandin) [inline lib/parallel_cucumber/worker.rb]: Use symbols.

Comment (nwlunatic) [reply lib/parallel_cucumber/worker.rb]: 👍

File: lib/parallel_cucumber/worker.rb (hunk 1 above)
Comment (nwlunatic) [reply lib/parallel_cucumber/worker.rb]: 👍

**Pull request merged**